### ISSUE TITLE
Angular Universal documentation missing files

### ISSUE THREAD

**Issue by LucasPaganini** [angular/angular#22873]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
[Angular Universal guide](https://angular.io/guide/universal#the-example) says that the final folder structure will look like this:
```
src/
  index.html                 app web page
  main.ts                    bootstrapper for client app
  main.server.ts             * bootstrapper for server app
  tsconfig.app.json          TypeScript client configuration
  tsconfig.server.json       * TypeScript server configuration
  tsconfig.spec.json         TypeScript spec configuration
  style.css                  styles for the app
  app/ ...                   application code
    app.server.module.ts     * server-side application module
server.ts                    * express web server
tsconfig.json                TypeScript client configuration
package.json                 npm configuration
webpack.server.config.js     * Webpack server configuration
```

But completing all the instructions leads to a project missing the `main.server.ts` file and also, the docs do not mention how to config the .angular-cli.json to have 2 apps in the "apps" array.
Without these files, the npm scripts to build the universal app do not work.

## Expected behavior
To have a functioning app after following all the instructions in the guide.

## Minimal reproduction of the problem with instructions
The bug can be reproduced following the instructions in the [Angular Universal guide](https://angular.io/guide/universal)

## What is the motivation / use case for changing the behavior?
Motivate new developers to use Angular by having a good documentation about Server Side Rendering.

## Environment
Angular CLI: 1.6.5
Node: 8.9.4
OS: darwin x64
Angular: 5.0.5
... animations, common, compiler, compiler-cli, core, forms
... http, platform-browser, platform-browser-dynamic
... platform-server, router, upgrade

@angular/cli: 1.6.5
@angular-devkit/build-optimizer: 0.0.42
@angular-devkit/core: 0.0.29
@angular-devkit/schematics: 0.0.52
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.5
@schematics/angular: 0.1.17
typescript: 2.4.2
webpack: 3.10.0

Browser: No browser.
 
For Tooling issues:
- Node version: 8.9.4
- Platform:  MAC darwin x64

Others: Nothing else

**Comment by LucasPaganini** [comment]
I managed to fix the errors, unfortunately, I wasn't using git while debugging so I can only provide a [working repo with one commit...](https://github.com/LucasPaganini/angular-universal-tour-of-heroes) :(

What I did: 
- Added a new app to "apps" array in the .angular-cli.json
- Added the main.server.ts file
- Downgraded the ts-loader webpack plugin from 4.1.0 to 2.3.7
- Added a .gitignore

Also: 
- I added a console.log in the server.ts file to measure the time spent rendering each requested route, that can be deleted.
- There's a[ part in the guide talking about absolute paths](https://angular.io/guide/universal#absolute-http-urls) that causes a bug because the code injects an `@Optional()` token in the HeroesService constructor and tries to use it in the beginning of a string. The problem is that the DI will return null if the token is not found, so the string would be "nullapi/heroes".

This should be added to the [official guide doc](https://angular.io/guide/universal).
